curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md

```diff
@@ -0,0 +1,147 @@
+---
+id: 66f686b8ebdb982fa8e14330
+title: Build a Date Conversion Program
+challengeType: 14
+dashedName: lab-date-conversion
+---
+
+# --description--
+
+In this lab, you will code a date conversion program that converts a given date to different formats. For example, the current date `Fri Sep 27 2024 16:04:43 GMT+0500 (Pakistan Standard Time)` would be formatted in the following 2 ways:
+
+- `(MM/DD/YYYY): 9/27/2024`.
+- `(Month Day, Year): September 27, 2024`.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should create a variable named `currentDate` and assign it the current date and time using the `Date` object.
+2. You should create a variable named `currentDateFormat` and assign it the string `Current Date and Time: [current date]`. Replace `[current date]` with the result of the `currentDate` variable.
+3. You should log the value of `currentDateFormat` to the console.
+4. You should create a function named `formatDateMMDDYYYY` that takes the date object as a parameter. You can name this parameter whatever you like.
+5. Your `formatDateMMDDYYYY` function should convert the current date to `month/day/year` format and return the string `Formatted Date (MM/DD/YYYY): [month]/[day]/[year]`.
+6. You should create a function named `formatDateLong` that takes the date object as a parameter.
+7. Your `formatDateLong` function should convert the current date to `Month Day, Year` format and return the string `Formatted Date (Month Day, Year): [formatted date]`.
+
+# --hints--
+
+You should have a variable named `currentDate` that holds the current date and time using the `Date` object.
+
+```js
+
+// excluded timestamps
+const expectedDate = new Date();
+const expectedDateOnly = new Date(expectedDate.getFullYear(), expectedDate.getMonth(), expectedDate.getDate());
+
+const currentDateOnly = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate());
+
+assert.equal(currentDate.toString(), expectedDate.toString());
+```
+
+You should have a variable named `currentDateFormat` that holds the current date in the format `Current Date and Time: [current date]`.
+
+```js
+const expectedDate = new Date();
+const expectedDateFormat = `Current Date and Time: ${expectedDate}`;
+
+assert.equal(currentDateFormat, expectedDateFormat);
+```
+
+You should log the value of `currentDateFormat` to the console.
+
+```js
+assert.match(code, /console\.log\(\s*currentDateFormat\s*\)/);
+```
+
+You should have a function `formatDateMMDDYYYY` 
+
+```js
+assert.isFunction(formatDateMMDDYYYY);
+```
+
+The function`formatDateMMDDYYYY` should take a date object as a parameter.
+
+```js
+assert.match(formatDateMMDDYYYY.toString(), /\s*formatDateMMDDYYYY\(\s*\w+\s*\)/);
+```
```

Comment:
this is not testing that the parameter is a date object. To test if it has one parameter you can check `formatDateMMDDYYY.length` (I learned recently that the lenght is equal to the number of parameters)

````suggestion
The function`formatDateMMDDYYYY` should take one parameter.

```js
assert.lengthOf(formatDateMMDDYYYY, 1);
```
````